pyproject.toml

```diff
@@ -34,11 +34,11 @@ scipy = ">=1.5.0"
 wtpsplit-lite = ">=0.1.0"
 # Large Language Models:
 huggingface-hub = ">=0.22.0"
-litellm = ">=1.48.4,<1.56.10"
+litellm = ">=1.48.4,<1.56.2"
 llama-cpp-python = ">=0.3.2"
 pydantic = ">=2.7.0"
 # Approximate Nearest Neighbors:
-pynndescent = ">=0.5.12"
+pynndescent = ">=0.5.13"
```

Comment:
What I think we want to do here is:
1. Explicitly upgrade ruff, mypy, and other tools in pyproject.toml.
2. Then run `poetry lock --no-update` to avoid updating any dependencies that we don't explicitly upgrade.

Running a full `poetry lock` would significantly change the dependency set we run the tests against.

In the future, we should test against both the 'oldest' solution to the specification, and the 'newest' solution to the specification. That's something we can do when we migrate from Poetry to uv.